File: api/ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/ContractVerificationFailure.java
package net.corda.v5.ledger.utxo;

import net.corda.v5.base.annotations.CordaSerializable;
import net.corda.v5.base.annotations.DoNotImplement;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Defines a contract verification failure.
 */
@DoNotImplement
@CordaSerializable
public interface ContractVerificationFailure {

    /**
     * Gets the class name of the {@link Contract} that caused verification failure.
     *
     * @return Returns the class name of the {@link Contract} that caused verification failure.
     */
    @NotNull
    String getContractClassName();

    /**
     * Gets the class names of the contract states that import the {@link Contract} that caused verification failure.
     *
     * @return Returns the class names of the contract states that import the {@link Contract} that caused verification failure.
     */
    @NotNull
    List<String> getContractStateClassNames();

    /**
     * Gets the class name of the {@link Exception} that caused verification failure.
     *
     * @return Returns the class name of the {@link Exception} that caused verification failure.
     */
    @NotNull
    String getExceptionClassName();

    /**
     * Gets the details of the {@link Exception} that caused verification failure.
     *
     * @return Returns the details of the {@link Exception} that caused verification failure.
     */
    @NotNull
    String getExceptionMessage();
}

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractVerificationFailure.kt
package net.corda.v5.ledger.utxo

import net.corda.v5.base.annotations.CordaSerializable
import net.corda.v5.base.annotations.DoNotImplement

/**
 * Defines a contract verification failure.
 *
 * @property contractClassName The class name of the [Contract] that caused verification failure.
 * @property contractStateClassNames The class names of the contract states that import the [Contract] that caused verification failure.
 * @property exceptionClassName The class name of the [Exception] that caused verification failure.
 * @property exceptionMessage The details of the [Exception] that caused verification failure.
 */
@CordaSerializable
@DoNotImplement
interface ContractVerificationFailure {
    val contractClassName: String
    val contractStateClassNames: List<String>
    val exceptionClassName: String
    val exceptionMessage: String
}
